### DISCUSSION TITLE
Image stays in upload folder

### DISCUSSION THREAD

**Discussion by wardwygaerts** [immich-app/immich#7908] [Q&A]
I dragged some new pictures into the webinterface and all files are handled correctly and moved to `upload/library`, but one file stays in `upload/upload`.
Metadata and thumbnail is generated correctly. If I remove the image (+ trash) and reupload, same thing happens.

How can I resolve this?

**Comment by alextran1502** [comment]
check the log, probably need to turn on debug mode (you can do this in Administration settings) and reupload the file after removing it from the server. I think the file doesn't have EXIF to be extracted to move to the correct location

**Comment by wardwygaerts** [comment]
Thanks for helping!
I see this error `QueryFailedError: duplicate key value violates unique constraint "UQ_newPath"`. But this image is not yet in my gallery.


See the log below:
`03/12/2024 08:42:15 PM [Nest] 6  - 03/12/2024, 8:42:15 PM   ERROR [StorageTemplateService] Problem applying storage template
03/12/2024 08:42:15 PM [Nest] 6  - 03/12/2024, 8:42:15 PM   ERROR [StorageTemplateService] QueryFailedError: duplicate key value violates unique constraint "UQ_newPath"
03/12/2024 08:42:15 PM    at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
03/12/2024 08:42:15 PM    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
03/12/2024 08:42:15 PM    at async InsertQueryBuilder.execute (/usr/src/app/node_modules/typeorm/query-builder/InsertQueryBuilder.js:106:33)
03/12/2024 08:42:15 PM    at async SubjectExecutor.executeInsertOperations (/usr/src/app/node_modules/typeorm/persistence/SubjectExecutor.js:260:42)
03/12/2024 08:42:15 PM     at async SubjectExecutor.execute (/usr/src/app/node_modules/typeorm/persistence/SubjectExecutor.js:92:9)
03/12/2024 08:42:15 PM    at async EntityPersistExecutor.execute (/usr/src/app/node_modules/typeorm/persistence/EntityPersistExecutor.js:140:21)
03/12/2024 08:42:15 PM    at async StorageCore.moveFile (/usr/src/app/dist/domain/storage/storage.core.js:144:20)
03/12/2024 08:42:15 PM    at async /usr/src/app/dist/domain/storage-template/storage-template.service.js:138:17
03/12/2024 08:42:15 PM    at async /usr/src/app/dist/infra/repositories/database.repository.js:185:23
03/12/2024 08:42:15 PM [Nest] 6  - 03/12/2024, 8:42:15 PM   ERROR [StorageTemplateService] Object:
03/12/2024 08:42:15 PM {
03/12/2024 08:42:15 PM   "id": "dd2b3d62-9f95-448c-ae33-849aa0a918df",
03/12/2024 08:42:15 PM  "oldPath": "upload/upload/8e5b757c-5f63-4f70-a421-97a7a3d9b35b/f4/ce/f4ce8f99-2c40-4c27-b6e4-88bb92ff6175.jpg",
03/12/2024 08:42:15 PM  "newPath": "upload/library/ward/2024/2024-03/IMG20240305222723.jpg"
03/12/2024 08:42:15 PM }
03/12/2024 08:42:15 PM
03/12/2024 08:42:15 PM [Nest] 6  - 03/12/2024, 8:42:15 PM     LOG [MediaService] Successfully generated JPEG image thumbnail for asset dd2b3d62-9f95-448c-ae33-849aa0a918df
03/12/2024 08:42:16 PM [Nest] 6  - 03/12/2024, 8:42:16 PM     LOG [MediaService] Successfully generated WEBP image thumbnail for asset dd2b3d62-9f95-448c-ae33-849aa0a918df
03/12/2024 08:42:17 PM [Nest] 6  - 03/12/2024, 8:42:17 PM   DEBUG [PersonService] 0 faces detected in upload/thumbs/8e5b757c-5f63-4f70-a421-97a7a3d9b35b/dd/2b/dd2b3d62-9f95-448c-ae33-849aa0a918df.jpeg`

**Comment by wardwygaerts** [comment]
@alextran1502 can you assist how to solve this duplicate key? Can I remove if from the database, so it's not recognizzed as suplicate?

**Reply by alextran1502** [reply]
You can, also you can also check if there is this file at the mentioned location `upload/library/ward/2024/2024-03/IMG20240305222723.jpg`

**Reply by wardwygaerts** [reply]
The file is not present in the mentioned location.
In which table do I need to search to remove the duplicate key? I assume if I remove it from the `assets` table, it will not solve anything?

**Reply by wardwygaerts** [reply]
@alextran1502 
I'm still strugling with this one image which stays in de upload folder. How can I remove this duplicate key?

**Comment by MatrixTechno** [comment]
I'm having this same issue, did you end up solving it?

**Reply by wardwygaerts** [reply]
I had only one pictures with this behavior. I downloaded the picture, removed it (also from trash), renamed the downloaded one and reuploaded it.
